### DISCUSSION TITLE
New chain contract for ERC-6492 validatior

### DISCUSSION THREAD

**Discussion by coffeexcoin** [wevm/viem#2804] [Idea / Feature Request]
I would like to propose the addition of a new property: `chain.contracts.universalSignatureValidator` for a canonical deployed signature validation contract.

The current `validateHash` public action uses a createless deploy of a signature validator which returns a value indicating if the signature is valid - this is useful for ERC-6492 validation of counterfactual smart wallets. One issue with the current implementation is that some chains (notably zk stack) do not support returning a value from a deployment the way the deployless validator works (zk stack has a canonical contract factory which only ever returns the contract address)

My proposal is to add the `universalSignatureValidator` for chains that do not support createless deploy and allow the chain to designate a deployed version of the [UniversalSigValidator](https://github.com/wevm/viem/blob/main/contracts/src/deployless/DeploylessUniversalSigValidator.sol) as the instance to be used. If the `universalSignatureValidator` exists, the action will simply call that address as opposed to doing a createless deploy.

**Comment by jxom** [comment]
Open to a PR!

**Reply by coffeexcoin** [reply]
Thanks! PR up at https://github.com/wevm/viem/pull/2807
